hasura/graphql-engine
# ra-data-hasura: fix key update when response not an array

### Description
If we have a custom primary key defined when clicking on "Edit" button `response[0]` is not an array. So the the `map` function is `undefined`. I also changed `map` to `forEach` because that makes more sense because we are not returning anything from the loop.

The correct fix might be on the server. But I am not familiar with that. The reason I say that is because when we are _not_ using a custom primary key an array is always returned. I think the server is handling the missing `id` field differently.

### Affected components 
- ra-data-hasura

### Solution and Design
Correct the bug.

### Steps to test and verify
Click "Edit" after setting a custom primary key and it will error with `response[0].forEach is not a function`. With this change it will no longer give an error.
